src/states/node.rs

```diff
@@ -541,26 +538,38 @@ impl Node {
                              -> Result<(), RoutingError> {
         signed_msg.check_integrity()?;
 
-        if self.in_authority(&signed_msg.routing_message().dst) {
+        let in_authority = self.in_authority(&signed_msg.routing_message().dst);
+
+        if in_authority {
             self.send_ack(signed_msg.routing_message(), route);
         }
 
         if self.routing_msg_filter.filter_incoming(signed_msg.routing_message(), route) > 1 {
             return Err(RoutingError::FilterCheckFailed);
         }
 
-        let in_authority = self.in_authority(&signed_msg.routing_message().dst);
-        if !in_authority && self.respond_from_cache(signed_msg.routing_message(), route)? {
+        if in_authority {
+            // if the last hop is us and the destination is a group - we need to forward it to the
+            // rest of the group
```

Comment:
`RT::targets` actually only returns the whole group if the target group is our group, so this is necessary, but it could be better to modify `targets`.

This approach has less edge cases, though - we don't have to consider what happens when the sender is in the target group etc. It is also pretty much analogous to accumulating in the sending group - the accumulation and broadcast happen in respective groups and all the communication between groups happens between single nodes.